### ISSUE TITLE
Sending messages with CORS enabled raises a `400 Bad request` with `Null body` error

### ISSUE THREAD

**Issue by ppatierno** [strimzi/strimzi-kafka-bridge#726]
This issue is referred to discussion https://github.com/strimzi/strimzi-kafka-bridge/discussions/696

When CORS is enabled this way ...

```shell
#Enable CORS
http.cors.enabled=true
http.cors.allowedOrigins=*
http.cors.allowedMethods=GET,POST,PUT,DELETE,OPTIONS,PATCH
```

and we send an HTTP request for producing messages, we get an error:

```shell
curl -X POST http://localhost:8080/topics/my-topic -H 'content-type: application/vnd.kafka.json.v2+json' -d '{"records": [{"key": "key-1","value": "value-1"},{"key": "key-2","value": "value-2"}]}'
{"error_code":400,"message":"[Bad Request] Body application/vnd.kafka.json.v2+json parsing error: Null body"}
```

Disabling CORS makes it working fine instead.

**Comment by ppatierno** [comment]
After digging into Vert.x Web and OpenAPI3RouterBuilder component, it turned out that the reason is the following ...

When CORS is enabled on the bridge, we create a CORS handler which is set as `rootHandler` in the OpenAPI3RouterBuilder instance here:

https://github.com/strimzi/strimzi-kafka-bridge/blob/main/src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java#L169

The builder adds this handler to a global handlers list:

https://github.com/vert-x3/vertx-web/blob/master/vertx-web-openapi/src/main/java/io/vertx/ext/web/openapi/impl/OpenAPI3RouterBuilderImpl.java#L183

But when it's time to create a route where handling a body is needed (i.e. producing a message), it just skips the addition of the `BodyHandler` because the global handlers list is not empty (we added the CORS handler).

https://github.com/vert-x3/vertx-web/blob/master/vertx-web-openapi/src/main/java/io/vertx/ext/web/openapi/impl/OpenAPI3RouterBuilderImpl.java#L259

It means that when CORS is enabled, whatever POST request is made with a body ... handling the body is just skipped.

This is the reason why when it comes to the validation, the `Null body` error is raised.

**Comment by ppatierno** [comment]
Opening an issue on the Vert.x Web repo here https://github.com/vert-x3/vertx-web/issues/2320 to discuss with the authors.

**Comment by edubois10** [comment]
Hi @ppatierno , I have the same behaviour than the one you described but additionally the GET method is not working either when adding CORS. I had a working kafka bridge instance where I could produce message to a topic and consume the records from a topic using the curl command.

To produce messages to a topic I was using:
````
 http://my-bridge-kafka-playground.apps.devops.edubois-cloud.org/topics/my-topic \
        -H 'Content-type: application/vnd.kafka.json.v2+json' \
        -d '{
    "records": [
        {
            "key": "key-1",
            "value": "value-1"
        },
        {
            "key": "key-2",
            "value": "value-2"
        }
    ]}'

output: {"offsets":[{"partition":1,"offset":54},{"partition":1,"offset":55}]}%    
````
and to consume the message I was doing the following:
````
curl -X GET http://my-bridge-kafka-playground.apps.devops.edubois-cloud.org/consumers/bridge-quickstart-consumer-group/instances/bridge-quickstart-consumer/records \
  -H 'accept: application/vnd.kafka.json.v2+json'
output: [{"topic":"my-topic","key":"key-1","value":"value-1","partition":1,"offset":20},{"topic":"my-topic","key":"key-2","value":"value-2","partition":1,"offset":21}]
````

**Afterwards** I added CORS to my kafka bridge instance
````
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaBridge
metadata:
  name: my-bridge
  namespace: kafka-playground
spec:
  bootstrapServers: 'my-cluster-kafka-bootstrap:9092'
  http:
    cors:
      allowedMethods:
        - GET
        - POST
        - DELETE
        - ACCEPT
      allowedOrigins:
        - '*'
    port: 8080
  replicas: 1
````
And the two curl commands failed with the following output:

GET method: {"error_code":404,"message":"The specified consumer instance was not found."}%  

POST method: {"error_code":400,"message":"Bad Request"}%

I need the CORS to produce and consume data from my nuxtjs frontend demo app.

**Comment by ppatierno** [comment]
@edubois10 consider that the fix will be available in the future 0.24.0 release, so it's on the main branch only right now.
